daemon/stats.go

```diff
@@ -35,11 +35,6 @@ func (daemon *Daemon) ContainerStats(prefixOrName string, config *ContainerStats
 		return json.NewEncoder(config.OutStream).Encode(&types.Stats{})
 	}
 
-	updates, err := daemon.subscribeToContainerStats(container)
-	if err != nil {
-		return err
-	}
-
 	if config.Stream {
```

Comment:
Just wondering, by moving this down below are you also removing a possible race condition between the Write(nil) and the first 'update'?  Sure seems like it, so +1!